app/models/ems_refresh/parsers/azure.rb

```diff
@@ -0,0 +1,214 @@
+          :vendor           => "Microsoft",
+          :raw_power_state  => vm["powerStatus"],
+          :operating_system => process_os(vm),
+          :flavor           => series,
+          :location         => uid,
+          :hardware         => {
+            :disks    => [], # Filled in later conditionally on flavor
+            :networks => [], # Filled in later conditionally on what's available
+          },
+        }
+        new_result[:availability_zone] = fetch_az(az) unless az.nil?
+
+        populate_hardware_hash_with_disks(new_result[:hardware][:disks], vm)
+        populate_hardware_hash_with_series(new_result[:hardware], vm, series)
+        populate_hardware_hash_with_networks(new_result[:hardware][:networks], vm)
+
+        return uid, new_result
+      end
+
+      def fetch_az(availability_zone)
+        @data_index.fetch_path(:availability_zones, availability_zone.downcase)
+      end
+
+      def process_os(vm)
+        {
+          :product_name => guest_os(vm)
+        }
+      end
+
+      def guest_os(vm)
+        image_reference = vm.fetch_path('properties', 'storageProfile', 'imageReference')
+        image_reference['offer'] + " " + image_reference['sku'].gsub('-', ' ')
+      end
+
+      def populate_hardware_hash_with_disks(hardware_disks_array, vm)
+        data_disks = vm.fetch_path('properties', 'storageProfile', 'dataDisks')
+        data_disks.each do |disk|
+          disk_size      = disk['diskSizeGB'] * 1.gigabyte
+          disk_name      = disk['name']
+          disk_location  = disk['vhd']['uri']
+
+          add_instance_disk(hardware_disks_array, disk_size, disk_name, disk_location)
+        end
+      end
+
+      def add_instance_disk(disks, size, name, location)
+        super(disks, size, name, location, "azure")
+      end
+
+      def populate_hardware_hash_with_networks(hardware_networks_array, vm)
+        nics = vm.fetch_path('properties', 'networkProfile', 'networkInterfaces')
+
+        nics.each do |n|
+          n['properties'].each do |n_prop|
+            private_network = {
+              :ipaddress => n_prop['properties']['privateIPAddress'],
+              :hostname  => n_prop['name']
+            }.delete_nils
+
+            public_network = {
+              :ipaddress => n_prop['properties']['publicIPAddress'],
+              :hostname  => n_prop['name']
+            }.delete_nils
+
+            hardware_networks_array <<
+              private_network.merge(:description => "private") unless private_network.blank?
+            hardware_networks_array <<
+              public_network.merge(:description => "public")   unless public_network.blank?
+          end
+          hardware_networks_array.flatten!
+        end
+      end
+
+      def populate_hardware_hash_with_series(hardware_hash, vm, series)
+        return if series.nil?
+        hardware_hash[:numvcpus]      = series[:cpus]
+        hardware_hash[:logical_cpus]  = series[:cpus]
```

Comment:
Yeah ... this looks like the same problem as SCVMM.  Just set `logical_cpus` and then the UI knows how to display that.